src/MetaVariables.cpp

```diff
@@ -0,0 +1,284 @@
+#include "webserv.hpp"
+#include <string.h>
+#include <utility>
+
+MetaVariables::MetaVariables()
+{
+}
+
+MetaVariables::MetaVariables(const MetaVariables &other) : metaVars(other.metaVars)
+{
+}
+
+MetaVariables &MetaVariables::operator=(const MetaVariables &other)
+{
+	if (this != &other)
+	{
+		metaVars = other.metaVars;
+	}
+	return *this;
+}
+
+void MetaVariables::setVar(const std::string &key, const std::string &value)
+{
+	metaVars[key] = value;
+}
+
+std::string MetaVariables::getVar(const std::string &key) const
+{
+	std::map<std::string, std::string>::const_iterator it = metaVars.find(key);
+	if (it != metaVars.end())
+	{
+		return it->second;
+	}
+	else
+	{
+		return "";
+	}
+}
+
+std::vector<char *> MetaVariables::getForExecve() const
+{
+	std::vector<char *> result;
+	for (std::map<std::string, std::string>::const_iterator it = metaVars.begin(); it != metaVars.end(); ++it)
+	{
+		std::string env = it->first + "=" + it->second;
+		char *envCStr = new char[env.size() + 1];
+		size_t i;
+		for (i = 0; i < env.size(); ++i)
+		{
+			envCStr[i] = env[i];
+		}
+		envCStr[i] = '\0';
+		result.push_back(envCStr);
+	}
+	result.push_back(NULL);
+	return result;
+}
+
+std::pair<std::string, std::string> MetaVariables::separatePathAndInfo(const std::string &requestTarget) const
+{
+	const char *extensions[] = {".cgi", ".pl", ".py", ".php"};
+	const size_t extCount = sizeof(extensions) / sizeof(extensions[0]);
+
+	std::string scriptPath;
+	std::string pathInfo;
+
+	size_t pos = std::string::npos;
+	for (size_t i = 0; i < extCount; ++i)
+	{
+		pos = requestTarget.find(extensions[i]);
+		if (pos != std::string::npos)
+		{
+			pos += strlen(extensions[i]);
+			break;
+		}
+	}
+
+	if (pos != std::string::npos)
+	{
+		scriptPath = requestTarget.substr(0, pos);
+		if (pos < requestTarget.length())
+			pathInfo = requestTarget.substr(pos);
+	}
+	else
+	{
+		scriptPath = requestTarget;
+	}
+
+	return std::make_pair(scriptPath, pathInfo);
+}
+
+bool startsWith(const std::string &str, const std::string &prefix)
+{
+	if (str.length() < prefix.length())
+	{
+		return false;
+	}
+	for (std::string::size_type i = 0; i < prefix.length(); ++i)
+	{
+		if (str[i] != prefix[i])
+		{
+			return false;
+		}
+	}
+	return true;
+}
+
+// bool MetaVariables::isAuthorityForm(const HTTPRequest &request)
```

Comment:
Do you need all this commented out part?

Reply:
https://github.com/dantol29/webserver/pull/117/files/19b7efdf5c56ab1f0617af11a3ffb949d4f90b95#r1569252302